From e2063e3893b01268c5821b47a663334e0ecc8c4e Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 4 Jun 2025 22:46:52 +0100
Subject: [PATCH 3/4] test_analyze_wheel_abi: reload and reimport wheel_abi

It seems that in addition to reloading lddtree, reload of
auditwheel.wheel_abi is needed to full flush and reset
analyze_wheel_abi() lru_caches.
---
 tests/integration/test_bundled_wheels.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tests/integration/test_bundled_wheels.py b/tests/integration/test_bundled_wheels.py
index 0c1d2227..7fa01700 100644
--- a/tests/integration/test_bundled_wheels.py
+++ b/tests/integration/test_bundled_wheels.py
@@ -11,6 +11,7 @@
 
 import pytest
 
+import auditwheel.wheel_abi
 from auditwheel import lddtree, main_repair
 from auditwheel.architecture import Architecture
 from auditwheel.libc import Libc
@@ -87,7 +88,8 @@ def test_analyze_wheel_abi(file, external_libs, exclude, env):
     with pytest.MonkeyPatch.context() as cp:
         if env:
             cp.setenv(env, f"{HERE}")
-            importlib.reload(lddtree)
+        importlib.reload(lddtree)
+        importlib.reload(auditwheel.wheel_abi)
 
         winfo = analyze_wheel_abi(
             Libc.GLIBC, Architecture.x86_64, HERE / file, exclude, False, True
@@ -96,8 +98,8 @@ def test_analyze_wheel_abi(file, external_libs, exclude, env):
             f"{HERE}, {exclude}, {env}"
         )
 
-    if env:
-        importlib.reload(lddtree)
+    importlib.reload(lddtree)
+    importlib.reload(auditwheel.wheel_abi)
 
 
 def test_analyze_wheel_abi_pyfpe():
